statics/workflows/flow-validation.yaml

```diff
@@ -92,38 +97,66 @@ Source: |
       capture = client.captures.create(capture)
       var timediff = interval * 1000;
       sleep(timediff)
-      var flowExpected = client.gremlin.G().V().Has('Captures.ID', capture.UUID, 'Type', NE('ovsport')).result();
-      var flowCaptured = client.gremlin.G().Flows().Has('CaptureID', capture.UUID).CaptureNode().Dedup().result();
-      var noflows = [];
 
-      if (protocol != "udp" && flowExpected.length && flowExpected.length == flowCaptured.length) {
-        flowCaptured = client.gremlin.G().Flows().Has('CaptureID', capture.UUID, 'Metric.ABPackets', GT(0), 'Metric.BAPackets', GT(0)).CaptureNode().Dedup().result();
+      var captureValidation = function(Expected) {
+        var noCapture = [];
+        for (var i = 0; i != Expected.length; i++) {
+          if (Expected[i].Metadata.State == "DOWN") {
+            noCapture.push(Expected[i]);
+          }
+        }
+        return noCapture
       }
 
-      if (flowExpected.length != flowCaptured.length) {
-        for (var i = 0; i != flowExpected.length; i++) {
+      var flowValidation = function(Expected, Captured) {
+        var noFlows = [];
+        for (var i = 0; i != Expected.length; i++) {
           var found = false;
 
-          for (var j = 0; j != flowCaptured.length; j++) {
-            if (flowExpected[i].ID === flowCaptured[j].ID) {
+          for (var j = 0; j != Captured.length; j++) {
+            if (Expected[i].ID === Captured[j].ID) {
               found = true;
               break;
             }
           }
 
           if (!found) {
-            noflows.push(flowExpected[i]);
+            noFlows.push(Expected[i]);
           }
         }
+        return noFlows
       }
 
-      if (!flowExpected.length) {
-        result["Error"] = "No node with capture"
-      } else if (noflows.length == 0) {
-        result["Connectivity"] = true
-      } else {
+      var noCapture = []
+      var flowExpected = client.gremlin.G().V().Has('Captures.ID', capture.UUID, 'Type', NE('ovsport')).result();
+
+      if (flowExpected.length != captureExpected.length) {
+        noCapture = captureValidation(captureExpected)
+      }
+
+      if (noCapture.length != 0) {
         result["Connectivity"] = false
-        result["NotReachedNodes"] = noflows
+        result["NotCapturedNodes"] = noCapture
+      } else {
```

Comment:
I would always return `NotReachedNodes` merging noCapture and noFlows and I would return NotCapturedNodes as an information. So it will be like

```
result['NotCapturedNodes'] = noCapture
result['NotReachNodes'] = noFlows + noCapture
```

In fact `DOWN` interface didn't see the flow thus "not reached"

Reply:
Yes, we can merge noCapture with noFlows.
Sounds Good!